api/src/main/java/org/infinispan/api/annotations/indexing/Id.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.api.annotations.indexing;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.hibernate.search.engine.environment.bean.BeanRetrieval;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMapping;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.PropertyMappingAnnotationProcessorRef;
+import org.infinispan.api.common.annotations.indexing.IdProcessor;
+
+/**
+ * Maps a property to the identifier of documents in the index.
+ * <p>
+ * This annotation is only taken into account on {@link Indexed} types.
+ * <p>
+ * Simplified version for Infinispan of {@link org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId}
+ */
+@Documented
+@Target({ElementType.METHOD, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@PropertyMapping(processor = @PropertyMappingAnnotationProcessorRef(type = IdProcessor.class, retrieval = BeanRetrieval.CONSTRUCTOR))
+public @interface Id {
```

Comment:
Are you sure this is necessary in Infinispan? I thought the ID was always the cache key?